lib/parse-args.js

```diff
@@ -194,6 +194,52 @@ function buildYargs (withCommands = false) {
   return yargs
 }
 
+function loadConfigFile (path) {
+  let config = {}
+  const jsExts = ['.js', '.cjs']
+  const ymlExts = ['.yml', '.yaml']
+  const fileName = basename(path)
+
+  const ext = extname(path).toLowerCase()
+  if (jsExts.includes(ext)) {
+    config = require(path)
+  } else if (ymlExts.includes(ext)) {
+    config = require('js-yaml').load(readFileSync(path, 'utf8'))
+  } else if (ext === '.json' || fileName.slice(-2) === 'rc') {
+    config = JSON.parse(readFileSync(path, 'utf8'))
+  }
```

Comment:
I am going to put this into a try/catch block for the following reasons

1.  If a not properly formated json or yaml object is returned it will produce an error.
2. If the path doesn't exist, it will produce an error.
3.  It could pass an invalid configuration key is pass, it will essentially get ignored.

It could in the future be helpful to have a config schema validator.  Going to add unit tests for two of the potential issues above.

Reply:
I agree on a schema validator. It would help people a lot.

However masking erroneous configuration files by just ignoring them, or even just logging the problem and proceeding with defaults, is an antipattern to me. It results in unexpected behavior since c8 proceeds with different settings than intended.

Reply:
In my [PR to this PR](https://github.com/mcknasty/c8/pull/5) I did wrap some of these in try-catch, but I only caught specific errors, re-thowing the others. The idea was to enhance the message in those cases as the underlying error was obtuse. However I think that other errors, such as file not found, are generally acceptable to leave alone.